contracts/cw-proposal-single/src/proposal.rs

```diff
@@ -308,18 +332,199 @@ mod test {
             threshold.clone(),
             votes.clone(),
             Uint128::new(15),
-            false
+            false,
+            false,
         ));
         // Proposal being expired should not effect those results.
         assert!(check_is_passed(
             threshold.clone(),
             votes.clone(),
             Uint128::new(15),
-            true
+            true,
+            false
         ));
 
         // More votes == higher threshold => not passed.
-        assert!(!check_is_passed(threshold, votes, Uint128::new(17), false));
+        assert!(!check_is_passed(
+            threshold,
+            votes,
+            Uint128::new(17),
+            false,
+            false
+        ));
+    }
+
+    #[test]
+    fn test_revoting_majority_no_pass() {
+        // Revoting being allowed means that proposals may not be
+        // passed or rejected before they expire.
+        let threshold = Threshold::AbsolutePercentage {
+            percentage: PercentageThreshold::Majority {},
+        };
+        let votes = Votes {
+            yes: Uint128::new(7),
+            no: Uint128::new(4),
+            abstain: Uint128::new(2),
+        };
+
+        // 15 total votes. 7 yes and 2 abstain. Majority threshold. This
+        // should pass but revoting is enabled.
+        assert!(!check_is_passed(
+            threshold.clone(),
+            votes.clone(),
+            Uint128::new(15),
+            false,
+            true,
+        ));
+        // Proposal being expired should cause the proposal to be
+        // passed as votes may no longer be cast.
+        assert!(check_is_passed(
+            threshold,
+            votes,
+            Uint128::new(15),
+            true,
+            true
+        ));
+    }
+
+    #[test]
+    fn test_revoting_majority_rejection() {
+        // Revoting being allowed means that proposals may not be
+        // passed or rejected before they expire.
+        let threshold = Threshold::AbsolutePercentage {
+            percentage: PercentageThreshold::Majority {},
+        };
+        let votes = Votes {
+            yes: Uint128::new(4),
+            no: Uint128::new(7),
+            abstain: Uint128::new(2),
+        };
+
+        // Not expired, revoting allowed => no rejection.
+        assert!(!check_is_rejected(
+            threshold.clone(),
+            votes.clone(),
+            Uint128::new(15),
+            false,
+            true
+        ));
+
+        // Expired, revoting allowed => rejection.
+        assert!(check_is_rejected(
+            threshold,
+            votes,
+            Uint128::new(15),
+            true,
+            true
+        ));
+    }
+
+    #[test]
+    fn test_absolute_threshold() {
```

Comment:
These are great! A few comments on the test cases would be nice to have.

Reply:
Agreed on this front but great coverage